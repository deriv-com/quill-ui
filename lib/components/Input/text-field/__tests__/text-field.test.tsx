import { act, fireEvent, render, screen } from "@testing-library/react";
import TextField, { TextFieldProps } from "..";
import userEvent from "@testing-library/user-event";
import React from "react";

describe("TextField", () => {
    const setup = (props: Partial<TextFieldProps> = {}) => {
        const utils = render(<TextField {...props} />);
        const input = utils.getByRole("textbox");
        return {
            input,
            ...utils,
        };
    };

    it("should match snapshot", () => {
        const { container } = setup({ placeholder: "Enter text" });
        expect(container).toMatchSnapshot();
    });

    it("should render with default props", () => {
        setup();
        expect(screen.getByRole("textbox")).toBeInTheDocument();
    });

    it("should render with a label", () => {
        setup({ label: "Username" });
        expect(screen.getByLabelText("Username")).toBeInTheDocument();
    });

    it("should call onChange handler when input changes", () => {
        const handleChange = jest.fn();
        const { input } = setup({ onChange: handleChange });
        act(() => {
            fireEvent.change(input, { target: { value: "Hello" } });
        });
        expect(handleChange).toHaveBeenCalledTimes(1);
    });

    it("should show character counter if enabled", () => {
        setup({ show_counter: true, maxLength: 10 });
        act(() => {
            fireEvent.change(screen.getByRole("textbox"), {
                target: { value: "Hello" },
            });
        });
        expect(screen.getByText("5/10")).toBeInTheDocument();
    });

    it("should render disabled input", () => {
        setup({ disabled: true });
        expect(screen.getByRole("textbox")).toBeDisabled();
    });

    it("should handle the hover for outline variant", async () => {
        const onHover = jest.fn();
        setup({
            placeholder: "Placeholder",
            onMouseEnter: onHover,
            variant: "outline",
        });
        const input = screen.getByPlaceholderText("Placeholder");
        await userEvent.hover(input);
        expect(onHover).toHaveBeenCalledTimes(1);
        expect(input.parentElement).toMatchSnapshot();
    });

    it("should render with a left icon", () => {
        const { container } = setup({ leftIcon: <span>Left Icon</span> });
        expect(screen.getByText("Left Icon")).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });

    it("should display message when provided", () => {
        const { container } = setup({
            message: "Error message",
            status: "error",
        });
        expect(screen.getByText("Error message")).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });
});
