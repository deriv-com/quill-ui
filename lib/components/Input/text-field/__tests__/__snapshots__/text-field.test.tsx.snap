// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField should display message when provided 1`] = `
<div>
  <div
    class="quill-input__container"
  >
    <div
      class="quill-input__wrapper quill-input__wrapper__variant--outline quill-input__wrapper__variant--outline--error quill-input__wrapper__size--md"
    >
      <div
        class="quill-input-icons__wrapper"
      >
        <div
          class="quill-input-no-label__wrapper"
        >
          <input
            class="input peer input__align--left input__size--md"
            placeholder=""
            value=""
          />
        </div>
        <span
          class="icon_wrapper status-icon--error"
        >
          <svg
            height="24"
            role="img"
            viewBox="0 0 32 32"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.688 9.82 7.953 22.516c-.078.117-.078.195-.078.312 0 .313.234.547.547.547h15.117c.313 0 .586-.234.586-.547 0-.117-.04-.195-.117-.312L16.273 9.82A.29.29 0 0 0 16 9.625c-.156 0-.234.078-.312.195m-1.602-.976A2.25 2.25 0 0 1 16 7.75c.742 0 1.484.43 1.875 1.094l7.734 12.695c.235.39.391.82.391 1.29 0 1.327-1.094 2.421-2.46 2.421H8.421A2.434 2.434 0 0 1 6 22.828q0-.703.352-1.289zm3.164 12.031c0 .703-.586 1.25-1.25 1.25-.703 0-1.25-.547-1.25-1.25 0-.664.547-1.25 1.25-1.25.664 0 1.25.586 1.25 1.25m-.312-7.187v3.75c0 .546-.43.937-.938.937-.547 0-.937-.39-.937-.937v-3.75c0-.508.39-.938.937-.938a.95.95 0 0 1 .938.938"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      class="message__container message__container--md message__container__status--error"
    >
      <span
        class="message__container__text"
      >
        Error message
      </span>
    </div>
  </div>
</div>
`;

exports[`TextField should handle the hover for outline variant 1`] = `
<div
  class="quill-input-no-label__wrapper"
>
  <input
    class="input peer input__align--left input__size--md"
    placeholder="Placeholder"
    value=""
  />
</div>
`;

exports[`TextField should match snapshot 1`] = `
<div>
  <div
    class="quill-input__container"
  >
    <div
      class="quill-input__wrapper quill-input__wrapper__variant--outline quill-input__wrapper__variant--outline--neutral quill-input__wrapper__size--md"
    >
      <div
        class="quill-input-icons__wrapper"
      >
        <div
          class="quill-input-no-label__wrapper"
        >
          <input
            class="input peer input__align--left input__size--md"
            placeholder="Enter text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`TextField should render with a left icon 1`] = `
<div>
  <div
    class="quill-input__container"
  >
    <div
      class="quill-input__wrapper quill-input__wrapper__variant--outline quill-input__wrapper__variant--outline--neutral quill-input__wrapper__size--md"
    >
      <div
        class="quill-input-icons__wrapper"
      >
        <span
          class="icon_wrapper"
        >
          <span>
            Left Icon
          </span>
        </span>
        <div
          class="quill-input-no-label__wrapper"
        >
          <input
            class="input peer input__align--left input__size--md"
            placeholder=""
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
